graphene_sqlalchemy/tests/models.py

```diff
@@ -75,10 +75,14 @@ class Reporter(Base):
     email = Column(String(), doc="Email")
     favorite_pet_kind = Column(PetKind)
     pets = relationship(
-        "Pet", secondary=association_table, backref="reporters", order_by="Pet.id"
+        "Pet",
+        secondary=association_table,
+        backref="reporters",
+        order_by="Pet.id",
+        lazy="selectin",
     )
-    articles = relationship("Article", backref="reporter")
-    favorite_article = relationship("Article", uselist=False)
+    articles = relationship("Article", backref="reporter", lazy="selectin")
+    favorite_article = relationship("Article", uselist=False, lazy="selectin")
```

Comment:
Why do we need to explicitly set `lazy="selectin"` here? AsyncSession does indeed only support `selectin`, but this would have side effects on the sync tests, right?

Reply:
Yes `AsyncSession` only supports `selectin` and `joined` the reason for that is that a lazy load would trigger IO ops and they would require to be awaited. 
There is a huge amount of discussion on the sqlalchemy GitHub as well: https://github.com/sqlalchemy/sqlalchemy/discussions?discussions_q=lazy+async and a guide on sqlalchemy to avoid those problems: https://docs.sqlalchemy.org/en/20/orm/extensions/asyncio.html#preventing-implicit-io-when-using-asyncsession
This has side effects on the synchronous tests yes. Instead of running the select statements when the attribute is accessed they are now ran when the parent object is selected from the database. That is the reason why there is now `graphene_sqlalachemy/tests/models.py` and `graphene_sqlalachemy/tests/models_batching.py`. This way the original behavior is maintained for the batching tests, where the difference in the modes would actually matter 😉

Reply:
Got it, makes sense!